reasoner_pydantic/upgrade.py

```diff
@@ -0,0 +1,194 @@
+from collections import defaultdict
+
+
+def upgrade_from_1p2(old_dict, result_source="ARA", result_method="default"):
+    def edge_conversion(kedge):
+        kedge = dict(kedge)
+        ksubject = kedge.get("subject")
+        kpredicate = kedge.get("predicate")
+        kobject = kedge.get("object")
+        kqualifiers = []
+        hashing_qualifiers = []
+        ksource = None
+        provenance_tree = []
+        knegated = kedge.get("negated", False)
+        kattributes = []
+        for attribute in kedge.get("attributes"):
+            if "biolink:original_knowledge_source" == attribute.get(
+                "attribute_type_id"
+            ):
+                ksource = attribute["value"]
+                attribute["attribute_type_id"] = "biolink:primary_knowledge_source"
```

Comment:
In the merging converter, this was done for simplicity's sake, but it might be a good idea to preserve the distinction between "original" and "primary" at least for now.